Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeSystemComponent.cpp

```diff
@@ -112,4 +124,54 @@ namespace AzToolsFramework
         return AzToolsFramework::IsInFocusSubTree(entityId, m_focusRoot);
     }
 
+    void FocusModeSystemComponent::OnEntityInfoUpdatedAddChildEnd(AZ::EntityId parentId, AZ::EntityId childId)
+    {
+        // If the parent's entityId is in the list, add the child.
+        if (auto iter = AZStd::find(m_focusedEntityIdList.begin(), m_focusedEntityIdList.end(), parentId);
+            iter != m_focusedEntityIdList.end())
```

Comment:
suggestion: Again this is something for another PR, but thinking about how large `m_focusedEntityIdList` might become, we might want to consider using a sorted list (we can use `equal_range` on to do this in `logN` time as opposed to `N` time). I think this could be a great area for benchmarks and with the test coverage we have it will make setting up these scenarios much easier to compare before/after improvements. There just is a case these things can all get quite slow if the entity count does get very high